### PR TITLE
libhdcs.cpp patch

### PR DESCRIPTION
**Pull request by Ericyuanhui**
Signed-off-by: Ericyuanhui <yuanhui.xu@intel.com>

### CODE REVIEW

File: single_patch/librbd_hook_cpp/hdcs.cpp_librbd.diff.patch [hunk 1]
```diff
@@ -0,0 +1,2269 @@
...
+--- a/src/include/libhdcs.hpp
++++ b/src/include/libhdcs.hpp
+@@ -13,15 +13,15 @@ namespace hdcs {
+ typedef void (*callback_t)(int r, void *arg);
+ typedef void* hdcs_completion_t;
+ 
+-static void hdcs_aio_release(hdcs_completion_t c);
+-static int hdcs_aio_create_completion(void *cb_arg, callback_t complete_cb, hdcs_completion_t *c);
++void hdcs_aio_release(hdcs_completion_t c);
++int hdcs_aio_create_completion(void *cb_ar
```
Comment (zhouyuan) [inline single_patch/librbd_hook_cpp/hdcs.cpp_librbd.diff.patch]: I think we need to decided on the param here: rbd name or the new io_ctx

Comment (zhouyuan) [inline single_patch/librbd_hook_cpp/hdcs.cpp_librbd.diff.patch]: we could split this out and create a separate patch

**Comment by zhouyuan** [comment]
Can you also check if it's difficult to use the C API?

**Comment by Ericyuanhui** [comment]
close and split two part
